promql/engine.go

```diff
@@ -1144,6 +1145,7 @@ func (ev *evaluator) eval(expr parser.Expr) parser.Value {
 				mint := maxt - selRange
 				// Evaluate the matrix selector for this series for this step.
 				points = ev.matrixIterSlice(it, mint, maxt, points)
+				ev.currentSamples = tmpEvalPoints
```

Comment:
Not quite, the points are still in memory.

We probably want to rethink how matrixIterSlice tracking works - maybe make it external to matrixIterSlice?